src/DynamoCore/Graph/Annotations/AnnotationModel.cs

```diff
@@ -305,43 +300,15 @@ internal void UpdateBoundaryFromSelection()
                 {
                     X = regionX,
                     Y = regionY,
-                    Width = xDistance + maxWidth + ExtendSize,
-                    Height = yDistance + maxHeight + ExtendSize
+                    Width = xDistance + ExtendSize,
+                    Height = yDistance + ExtendSize + ExtendYHeight
                 };
              
                 this.X = region.X;              
                 this.Y = region.Y;
                 this.Width = Math.Max(region.Width, TextMaxWidth + ExtendSize);
                 this.Height = region.Height;
 
-                //Calculate the boundary if there is any overlap
-                ModelBase overlap = null;
-                foreach (var nodesList in Nodes)
-                {
-                    if (!region.Contains(nodesList.Rect))
-                    {
-                        overlap = nodesList;
-                        if (overlap.Rect.Top < this.X ||
-                                    overlap.Rect.Bottom > region.Bottom) //Overlap in height - increase the region height
-                        {
-                            if (overlap.Rect.Bottom - region.Bottom > 0)
-                            {
-                                this.Height += overlap.Rect.Bottom - region.Bottom + ExtendSize + ExtendYHeight;
```

Comment:
On the other hand here, the code here doesn't check if the `nodesList` (which should be a single node) is a `NodeModel` or not, so maybe that check was not even needed.

Reply:
> I left some comments, but I think the solution looks good to the best of my knowledge. Are the tests going to be added in a different PR?

Yes, the tests will be added on a different PR, I wanted to resolve this first to know if i have to add any new tests